esi_leap/objects/owner_change.py

```diff
@@ -0,0 +1,185 @@
+from esi_leap.common import statuses
+from esi_leap.common import utils
+from esi_leap.db import api as dbapi
+from esi_leap.objects import base
+from esi_leap.objects import fields
+from esi_leap.objects import lease as lease_obj
+from esi_leap.objects import offer as offer_obj
+from esi_leap.resource_objects import resource_object_factory as ro_factory
+
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+@versioned_objects_base.VersionedObjectRegistry.register
+class OwnerChange(base.ESILEAPObject):
+    dbapi = dbapi.get_instance()
+
+    fields = {
+        'id': fields.IntegerField(),
+        'uuid': fields.UUIDField(),
+        'from_owner_id': fields.StringField(),
+        'to_owner_id': fields.StringField(),
+        'resource_type': fields.StringField(),
+        'resource_uuid': fields.StringField(),
+        'start_time': fields.DateTimeField(nullable=True),
+        'end_time': fields.DateTimeField(nullable=True),
+        'fulfill_time': fields.DateTimeField(nullable=True),
+        'expire_time': fields.DateTimeField(nullable=True),
+        'status': fields.StringField(),
+    }
+
+    @classmethod
+    def get(cls, owner_change_uuid, context=None):
+        db_owner_change = cls.dbapi.owner_change_get_by_uuid(owner_change_uuid)
+        if db_owner_change:
+            return cls._from_db_object(context, cls(), db_owner_change)
+
+    @classmethod
+    def get_all(cls, filters, context=None):
+        db_owner_changes = cls.dbapi.owner_change_get_all(filters)
+        return cls._from_db_object_list(context, db_owner_changes)
+
+    def create(self, context=None):
+        updates = self.obj_get_changes()
+
+        @utils.synchronized(
+            utils.get_resource_lock_name(
+                updates['resource_type'], updates['resource_uuid']),
+            external=True)
+        def _create_owner_change():
+
+            if updates['start_time'] >= updates['end_time']:
+                raise exception.InvalidTimeRange(
+                    resource='owner_change',
+                    start_time=str(updates['start_time']),
+                    end_time=str(updates['end_time'])
+                )
+
+            ro = ro_factory.ResourceObjectFactory.get_resource_object(
+                updates['resource_type'], updates['resource_uuid'])
+            ro.verify_availability(updates['start_time'],
+                                   updates['end_time'],
+                                   is_owner_change=True)
+
+            db_owner_change = self.dbapi.owner_change_create(updates)
+            self._from_db_object(context, self, db_owner_change)
+
+        _create_owner_change()
+
+    def fulfill(self, context=None):
+        @utils.synchronized(utils.get_resource_lock_name(self.resource_type,
+                                                         self.resource_uuid),
+                            external=True)
+        def _fulfill_owner_change():
+            LOG.info("Setting owner of %s %s to %s",
```

Comment:
I think it's better to contain from_owner_id in logs.

Reply:
I'll add it in!